src/components/SelectInput/SelectInput.test.jsx

```diff
@@ -0,0 +1,284 @@
+
+      expect(mockedHandleFocus).toBeCalledTimes(1);
+    });
+
+    test('it fires onBlur callback on blur', () => {
+      const mockedHandleBlur = jest.fn();
+
+      render(
+        <SelectInput
+          id="testId"
+          onBlur={mockedHandleBlur}
+          placeholder="Test Placeholder"
+          options={selectOptions}
+        />,
+      );
+
+      fireEvent.blur(screen.getByRole('textbox'));
+
+      expect(mockedHandleBlur).toBeCalledTimes(1);
+    });
+  });
+
+  describe('React-Select', () => {
+    describe('single select', () => {
+      test('it has no items selected on initial render', () => {
+        render(
+          <form data-testid="form">
+            <label htmlFor="iceCreamFlavors">Ice Cream Flavors</label>
+            <Select options={selectOptions} name="iceCreamFlavors" id="iceCreamFlavors" />
+          </form>,
+        );
+
+        expect(screen.getByTestId('form')).toHaveFormValues({ iceCreamFlavors: '' });
+      });
+
+      test('it sets the selected value for item that is selected', async () => {
+        render(
+          <form data-testid="form">
+            <label htmlFor="iceCreamFlavors">Ice Cream Flavors</label>
+            <Select options={selectOptions} name="iceCreamFlavors" id="iceCreamFlavors" />
+          </form>,
+        );
+        await selectEvent.select(screen.getByLabelText('Ice Cream Flavors'), ['Vanilla']);
+        expect(screen.getByTestId('form')).toHaveFormValues({ iceCreamFlavors: 'vanilla' });
+      });
+    });
+
+    describe('multi select', () => {
+      test('it has no items selected on initial render', () => {
+        render(
+          <form data-testid="form">
+            <label htmlFor="iceCreamFlavors">Ice Cream Flavors</label>
+            <Select options={selectOptions} name="iceCreamFlavors" id="iceCreamFlavors" isMulti />
+          </form>,
+        );
+
+        expect(screen.getByTestId('form')).toHaveFormValues({ iceCreamFlavors: '' });
+      });
+
+      test('it allows the selection of multiple items', async () => {
+        render(
+          <form data-testid="form">
+            <label htmlFor="iceCreamFlavors">Ice Cream Flavors</label>
+            <Select options={selectOptions} name="iceCreamFlavors" id="iceCreamFlavors" isMulti />
+          </form>,
+        );
+        await selectEvent.select(screen.getByLabelText('Ice Cream Flavors'), ['Chocolate', 'Vanilla']);
+        expect(screen.getByTestId('form')).toHaveFormValues({ iceCreamFlavors: ['chocolate', 'vanilla'] });
+
+        await selectEvent.select(screen.getByLabelText('Ice Cream Flavors'), 'Strawberry');
+        expect(screen.getByTestId('form'))
+          .toHaveFormValues({ iceCreamFlavors: ['chocolate', 'vanilla', 'strawberry'] });
+      });
+    });
+  });
+
+  describe('States', () => {
+    describe('No label, with a placeholder', () => {
+      test('it renders input without a label, and with a placeholder', () => {
```

Comment:
stylistically, do we need to have the nested describe with a single test description? Why not just have the test?

Reply:
Dunno. You say what you're describing, then describe it.

Reply:
I don't feel super strongly on this one way or the other, but I can see how nesting single tests can lead to inconsistency since it's not clear when to nest and when not to. I'd rather use the top-level `describe`  in the test itself and keep the `states` as the parent. The test copy itself seems a bit redundant in these cases.

Reply:
To clarify, I would probably do it like this:

```
describe('states', () => {
  test('With a label, and no custom placeholder', () => {
    // test here
  });
});
```

@JCarmine @nathanyoung